lhe2root/main.go

```diff
@@ -75,88 +77,84 @@ func main() {
 
 	// Loop over events
 	iEvt := 0
-	for i := 0; ; i++ {
+
+loop:
+	for {
 
 		// Decode this event, stop if the end of file is reached
 		lhe_evt, err := lhedec.Decode()
-		if err == io.EOF {
-			break
+		if err != nil {
+			if err == io.EOF {
+				break loop
+			}
+			log.Fatalf("could not decode event %d: %+v", iEvt, err)
 		}
 
 		// Print the event in verbose mode
 		if *verbose {
 			fmt.Println()
 			fmt.Println(*lhe_evt)
 		}
-		
+
 		// Converting the information from LHE event to TTree event
 		var (
 			pids     = lhe_evt.IDUP
 			PxPyPzEM = lhe_evt.PUP
-			get4Vec  = func(x [5]float64) (lv.FourVec) {
-				return lv.NewFourVecPxPyPzM(x[0], x[1], x[2], x[4])
-			}
 			setPart  = func(part *Particle, P lv.FourVec, pid int64) {
-				part.pt  = float32(P.Pt())
+				part.pt = float32(P.Pt())
 				part.eta = float32(P.Eta())
 				part.phi = float32(P.Phi())
-				part.m   = float32(P.M())
+				part.m = float32(P.M())
 				part.pid = int32(pid)
 			}
 		)
 
 		// Loop over particles
 		for i, pid := range pids {
-			if pid == 6 { // top-quark
+			switch int(pid) {
+			case heppdt.PDG_t:
 				setPart(&e.t, get4Vec(PxPyPzEM[i]), pid)
-			}
-			if pid == -6 { // anti top-quark
+			case heppdt.PDG_anti_t:
 				setPart(&e.tbar, get4Vec(PxPyPzEM[i]), pid)
-			}
-			if pid == 5 { // b-quark
+			case heppdt.PDG_b:
 				setPart(&e.b, get4Vec(PxPyPzEM[i]), pid)
-			}
-			if pid == -5 { // anti b-quark
+			case heppdt.PDG_anti_b:
 				setPart(&e.bbar, get4Vec(PxPyPzEM[i]), pid)
-			}
-			if pid == 24 { // W+ boson
+			case heppdt.PDG_W_plus:
 				setPart(&e.W, get4Vec(PxPyPzEM[i]), pid)
-			}
-			if pid == -24 { // W- boson
+			case heppdt.PDG_W_minus:
 				setPart(&e.Wbar, get4Vec(PxPyPzEM[i]), pid)
-			}
-			if (pid == -11 || pid == -13 || pid == -15) { // Charged leptons
+			case heppdt.PDG_e_plus, heppdt.PDG_mu_plus, heppdt.PDG_tau_plus:
 				setPart(&e.l, get4Vec(PxPyPzEM[i]), pid)
-			}
-			if (pid == 11 || pid == 13 || pid == 15) { // Charged anti-leptons
+			case heppdt.PDG_e_minus, heppdt.PDG_mu_minus, heppdt.PDG_tau_minus:
 				setPart(&e.lbar, get4Vec(PxPyPzEM[i]), pid)
-			}
-			if (pid == 10 || pid == 12 || pid == 14) { // Neutrinos
+			case heppdt.PDG_nu_e, heppdt.PDG_nu_mu, heppdt.PDG_nu_tau:
 				setPart(&e.v, get4Vec(PxPyPzEM[i]), pid)
-			}
-			if (pid == -10 || pid == -12 || pid == -14) { // Anti-neutrinos
+			case heppdt.PDG_anti_nu_e, heppdt.PDG_anti_nu_mu, heppdt.PDG_anti_nu_tau:
```

Comment:
AFAICT, neither `10` nor `-10` are valid PDG-IDs.
(I suspect you wanted `12,14,16` and `-12,-14,-16`)

Reply:
Good catch, thanks!